tests/qa-activity-report/generate-qa-activity-report.go

```diff
@@ -0,0 +1,201 @@
+package main
+
+import (
+	"bytes"
+	"encoding/csv"
+	"fmt"
+	"os"
+	"os/exec"
+	"strconv"
+	"strings"
+	"time"
+)
+
+func getGitLog(since, until string) ([]string, error) {
+	cmd := exec.Command("git", "log", "--since="+since, "--until="+until, "--pretty=format:%H %ad", "--date=short", "--", "tests/acceptance/features", ":(exclude)tests/acceptance/features/bootstrap/")
+	var out bytes.Buffer
+	cmd.Stdout = &out
+	if err := cmd.Run(); err != nil {
+		return nil, err
+	}
+	return strings.Split(out.String(), "\n"), nil
+}
+
+func getGitDiff(commit string) ([]string, error) {
+	cmd := exec.Command("git", "diff", commit+"~1", commit, "--", "tests/acceptance/features", ":(exclude)tests/acceptance/features/bootstrap/")
+	var out bytes.Buffer
+	cmd.Stdout = &out
+	err := cmd.Run()
+	if err != nil {
+		return nil, err
+	}
+	return strings.Split(out.String(), "\n"), nil
+}
+
+func generateCSV(csvRows [][]string, filePath string) error {
+	file, err := os.Create(filePath)
+	if err != nil {
+		return err
+	}
+	defer file.Close()
+
+	writer := csv.NewWriter(file)
+	defer writer.Flush()
+
+	header := []string{"Test-Type", "Date", "Tests Added", "Tests Changed", "Tests Deleted", "commit-ID"}
+	if err := writer.Write(header); err != nil {
+		return err
+	}
+	for _, row := range csvRows {
+		if err := writer.Write(row); err != nil {
+			return err
+		}
+	}
+
+	return nil
+}
+
+func main() {
+	monthStr := os.Getenv("MONTH")
+	yearStr := os.Getenv("YEAR")
+	daysStr := os.Getenv("DAYS")
+
+	var since, until, period string
+
+	if monthStr != "" && yearStr != "" {
+		month, err := strconv.Atoi(monthStr)
+		if err != nil {
+			fmt.Println("Invalid month:", err)
+			return
+		}
+		year, err := strconv.Atoi(yearStr)
+		if err != nil {
+			fmt.Println("Invalid year:", err)
+			return
+		}
+		startDate := time.Date(year, time.Month(month), 1, 0, 0, 0, 0, time.UTC)
+		endDate := startDate.AddDate(0, 1, -1)
+		since = startDate.Format("2006-01-02")
+		until = endDate.Format("2006-01-02")
+		period = fmt.Sprintf("%02d_%04d", month, year)
+	} else if daysStr != "" {
+		days, err := strconv.Atoi(daysStr)
+		if err != nil {
+			fmt.Println("Invalid number of days:", err)
+			return
+		}
+		until = time.Now().Format("2006-01-02")
+		since = time.Now().AddDate(0, 0, -days).Format("2006-01-02")
+		period = fmt.Sprintf("Last_%d_days", days)
+	} else {
+		fmt.Println("Please provide either MONTH and YEAR or DAYS.")
+		return
+	}
```

Comment:
Can we move all this to separate function `func getTimeframe() (until string, since string, period string)` (or similar, naming tbd)